tests/test_tutorials.py

```diff
@@ -1,176 +1,67 @@
-                return cell
-        except AttributeError:
-            continue
-
-    raise ValueError("no cell named '{}'".format(name))
-
-
-@pytest.mark.skipif(is_py2, reason=py2_deprecated)
-@pytest.mark.skipif(not jupyter_installed, reason=jupyter_required)
-def test_westeros_baseline(capsys):
-    fname = os.path.join(westeros_path, 'westeros_baseline.ipynb')
-    nb, errors = _notebook_run(fname, capsys=capsys)
-    assert errors == []
-
-    # On Python 2, the returned objective value is different
-    if sys.version_info[0] == 3:
-        return
-
-    cell = get_cell_by_name(nb, 'solve-objective-value')
-    obs = eval(cell['outputs'][0]['data']['text/plain'])
-    exp = 187445.953125
-    assert np.isclose(obs, exp)
-
-
-@pytest.mark.skipif(is_py2, reason=py2_deprecated)
-@pytest.mark.skipif(not jupyter_installed, reason=jupyter_required)
-def test_westeros_emissions(capsys):
-    fname = os.path.join(westeros_path, 'westeros_emissions_bounds.ipynb')
-    nb, errors = _notebook_run(fname, capsys=capsys)
-    assert errors == []
-
-
-@pytest.mark.skipif(is_py2, reason=py2_deprecated)
-@pytest.mark.skipif(not jupyter_installed, reason=jupyter_required)
-def test_westeros_emissions_tax(capsys):
-    fname = os.path.join(westeros_path, 'westeros_emissions_taxes.ipynb')
-    nb, errors = _notebook_run(fname, capsys=capsys)
-    assert errors == []
-
-
-@pytest.mark.skipif(is_py2, reason=py2_deprecated)
-@pytest.mark.skipif(not jupyter_installed, reason=jupyter_required)
-def test_westeros_firm_capacity(capsys):
-    fname = os.path.join(westeros_path, 'westeros_firm_capacity.ipynb')
-    nb, errors = _notebook_run(fname, capsys=capsys)
-    assert errors == []
-
-
-@pytest.mark.skipif(is_py2, reason=py2_deprecated)
-@pytest.mark.skipif(not jupyter_installed, reason=jupyter_required)
-def test_westeros_flexible_generation(capsys):
-    fname = os.path.join(westeros_path, 'westeros_flexible_generation.ipynb')
-    nb, errors = _notebook_run(fname, capsys=capsys)
-    assert errors == []
+    from pathlib2 import Path
+import sys
 
+from ixmp.testing import run_notebook, get_cell_output
+import numpy as np
+import pytest
 
-@pytest.mark.skipif(is_py2, reason=py2_deprecated)
-@pytest.mark.skipif(not jupyter_installed, reason=jupyter_required)
-def test_austria(capsys):
-    fname = os.path.join(ene_path, 'austria.ipynb')
-    nb, errors = _notebook_run(fname, capsys=capsys)
-    assert errors == []
 
+# Skip the entire file on Python 2
+if sys.version_info[0] == 2:
+    py2_deprecated = 'Python 2 is deprecated in the tutorials'
+    pytestmark = pytest.mark.skip(reason=py2_deprecated)
+
+
+AT = 'Austrian_energy_system'
+
+# Argument values to parametrize test_tutorial
+# Each item is a 3-tuple:
+# 1. Path fragments under tutorials directory,
+# 2. Name or index of cell containing objective value (or None),
+# 3. Expected objective value.
+tutorials = [
+    (('westeros', 'westeros_baseline'),
+     'solve-objective-value', 207544.09375),
```

Comment:
for future expansion, should the expected value actually be iterable here? I can imagine wanting to check multiple cells at some point. what do you think?

Reply:
Sure, easily done.